### ISSUE TITLE
Encountered an error "DEB file didn't have the control xattr" when running a convey build using GitHub Actions

### ISSUE THREAD

**Issue by guiyanakuang** [hydraulic-software/conveyor#114]
**Describe the bug**
```
Cache restored successfully
Cache restored from key: conveyor-9393781796
Run if [ ! -d ".conveyor/install/conveyor-14.2" ]; then
  if [ ! -d ".conveyor/install/conveyor-14.2" ]; then
    wget https://downloads.hydraulic.dev/conveyor/conveyor-14.2-linux-amd64.tar.gz
    mkdir -p .conveyor/install
    tar xzvf conveyor-14.2-linux-amd64.tar.gz -C .conveyor/install      
  fi
  shell: /usr/bin/bash --noprofile --norc -e -o pipefail {0}
  env:
    BUILD_FULL_PLATFORM: YES
    BUILD_MAIN_PATH: /home/runner/work/my-project/my-project/composeApp/build/classes/kotlin/desktop/main
    JAVA_HOME: /opt/hostedtoolcache/Java_Temurin-Hotspot_jdk/17.0.11-9/x64
    JAVA_HOME_17_X64: /opt/hostedtoolcache/Java_Temurin-Hotspot_jdk/17.0.11-9/x64
    APPLE_ASP: ***
    MAC_SIGNING_KEY_FILE_PATH: /home/runner/work/my-project/my-project/developerID_application.p12
    RELEASE_TOKEN: ***
Run PATH="$PATH:.conveyor/install/conveyor-14.2/bin"
  PATH="$PATH:.conveyor/install/conveyor-14.2/bin"
  conveyor --cache-dir=".conveyor/cache" "-Kapp.signing-key=$SIGNING_KEY" -f ci.conveyor.conf make site
  shell: /usr/bin/bash --noprofile --norc -e -o pipefail {0}
  env:
    BUILD_FULL_PLATFORM: YES
    BUILD_MAIN_PATH: /home/runner/work/my-project/my-project/composeApp/build/classes/kotlin/desktop/main
    JAVA_HOME: /opt/hostedtoolcache/Java_Temurin-Hotspot_jdk/17.0.11-9/x64
    JAVA_HOME_17_X64: /opt/hostedtoolcache/Java_Temurin-Hotspot_jdk/17.0.11-9/x64
    APPLE_ASP: ***
    MAC_SIGNING_KEY_FILE_PATH: /home/runner/work/my-project/my-project/developerID_application.p12
    RELEASE_TOKEN: ***
    SIGNING_KEY: ***
    CONVEYOR_AGREE_TO_LICENSE: 1
  0.00 secs │        │ Including config from Gradle
  5.27 secs │  done  │ Including config from Gradle
  8.28 secs │   0.0% │   0 /  65 │ Building
Warning: Could not connect to localhost:3000. Assuming this will be the first
release.
 17.82 secs │        │           │ 3. Base Configured Inputs for Linux Intel: Preparing to copy file tree
 17.82 secs │        │           │ 2. Base Configured Inputs for Windows Intel: Preparing to copy file tree
 17.93 secs │   0.0% │   0 /  17 │ 3. Base Configured Inputs for Linux Intel: Copying 17 file(s)
 17.93 secs │   0.0% │   0 /  18 │ 2. Base Configured Inputs for Windows Intel: Copying 18 file(s)
 18.39 secs │ 100.0% │  18 /  18 │ 2. Base Configured Inputs for Windows Intel: Copying 18 file(s)
 18.39 secs │   0.0% │   0 /   3 │ 2. Base Configured Inputs for Windows Intel: Copying directory permissions
 18.39 secs │ 100.0% │   3 /   3 │ 2. Base Configured Inputs for Windows Intel: Copying directory permissions
 19.60 secs │ 100.0% │ 39.01 MB / 39.01 MB │ 0. Debian Indexes: Downloading Contents-amd64.gz
 24.19 secs │        │                     │ 1. Base Configured Inputs for macOS Intel: Preparing to copy file tree
 24.19 secs │   0.0% │             0 /  17 │ 1. Base Configured Inputs for macOS Intel: Copying 17 file(s)
 24.19 secs │        │                     │ 0. Base Configured Inputs for macOS ARM: Preparing to copy file tree
 24.19 secs │   0.0% │             0 /  17 │ 0. Base Configured Inputs for macOS ARM: Copying 17 file(s)
 25.81 secs │ 100.0% │            17 /  17 │ 0. Base Configured Inputs for macOS ARM: Copying 17 file(s)
 25.81 secs │   0.0% │             0 /   3 │ 0. Base Configured Inputs for macOS ARM: Copying directory permissions
 25.81 secs │ 100.0% │             3 /   3 │ 0. Base Configured Inputs for macOS ARM: Copying directory permissions
 30.66 secs │ 100.0% │            17 /  17 │ 3. Base Configured Inputs for Linux Intel: Copying 17 file(s)
 30.66 secs │   0.0% │             0 /   3 │ 3. Base Configured Inputs for Linux Intel: Copying directory permissions
 30.66 secs │ 100.0% │             3 /   3 │ 3. Base Configured Inputs for Linux Intel: Copying directory permissions
 32.11 secs │   0.0% │             0 /   9 │ 0. Rendered icons for macOS: Resampling images
 32.95 secs │ 100.0% │             9 /   9 │ 0. Rendered icons for macOS: Resampling images
 33.63 secs │ 100.0% │            17 /  17 │ 1. Base Configured Inputs for macOS Intel: Copying 17 file(s)
 33.64 secs │   0.0% │             0 /   3 │ 1. Base Configured Inputs for macOS Intel: Copying directory permissions
 33.64 secs │ 100.0% │             3 /   3 │ 1. Base Configured Inputs for macOS Intel: Copying directory permissions
 33.67 secs │   0.0% │             0 / 136 │ 0. Processed Jars for Windows Intel: Repacking JARs
Warning: Application accesses internal Java APIs, inferring:
--add-opens=java.base/sun.security.x509=ALL-UNNAMED
 39.51 secs │   0.0% │             0 / 136 │ 1. Processed Jars for macOS Intel: Repacking JARs
 40.40 secs │ 100.0% │   8.54 MB / 8.54 MB │ 2. Mac Bundle Extras for ARM: Extracting Sparkle-2.5.1-hydraulic.tar.xz
 40.54 secs │ 100.0% │   8.54 MB / 8.54 MB │ 3. Mac Bundle Extras for Intel: Extracting Sparkle-2.5.1-hydraulic.tar.xz
 41.28 secs │   0.0% │             0 / 136 │ 3. Processed Jars for macOS ARM: Repacking JARs
 41.89 secs │ 100.0% │           136 / 136 │ 0. Processed Jars for Windows Intel: Repacking JARs
 41.89 secs │   0.0% │             0 / 136 │ 0. Processed Jars for Windows Intel: Scanning JARs
 45.27 secs │ 100.0% │           136 / 136 │ 1. Processed Jars for macOS Intel: Repacking JARs
 45.27 secs │   0.0% │             0 / 136 │ 1. Processed Jars for macOS Intel: Scanning JARs
 48.10 secs │ 100.0% │           136 / 136 │ 3. Processed Jars for macOS ARM: Repacking JARs
 48.10 secs │   0.0% │             0 / 136 │ 3. Processed Jars for macOS ARM: Scanning JARs
 50.62 secs │ 100.0% │           136 / 136 │ 1. Processed Jars for macOS Intel: Scanning JARs
 50.62 secs │ 100.0% │           136 / 136 │ 0. Processed Jars for Windows Intel: Scanning JARs
 50.81 secs │ 100.0% │           136 / 136 │ 3. Processed Jars for macOS ARM: Scanning JARs
 50.86 secs │ 100.0% │            65 /  65 │ Building
Internal error. Use --show-log for details.
hydraulic.factory.builds.BuildSystemScheduler$TaskFailure: Task failed: Apt Repository (apt-repository)
        at hydraulic.factory.builds.BuildSystemScheduler$executeTask$$inlined$executeWithNamedThreadAndLogging$1.call(Unknown Source)
        at hydraulic.utils.concurrency.Threading.runWithLoggingScope(Unknown Source)
        at hydraulic.factory.builds.BuildSystemScheduler.executeTask(Unknown Source)
        at hydraulic.factory.builds.BuildSystemScheduler.parallelScheduler$lambda$5(Unknown Source)
        at java.base/java.util.concurrent.CompletableFuture$AsyncSupply.run(CompletableFuture.java:1768)
        at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1136)
        at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:635)
        at hydraulic.factory.builds.BuildSystemScheduler$WorkerThread.run(Unknown Source)
Caused by: java.lang.IllegalStateException: DEB file didn't have the control xattr
        at hydraulic.conveyor.tasks.linux.AbstractAptRepositoryTask.build(Unknown Source)
        at hydraulic.factory.tasks.cached.CachedBuildTask$load$block$1.invoke(Unknown Source)
        at hydraulic.factory.tasks.cached.CachedBuildTask$load$block$1.invoke(Unknown Source)
        at hydraulic.factory.tasks.cached.CachedBuildTask.load$lambda$4(Unknown Source)
        at hydraulic.diskcache.LocalDiskCache.buildEntry(Unknown Source)
        at hydraulic.diskcache.LocalDiskCache.synchronizedBuild(Unknown Source)
        at hydraulic.diskcache.LocalDiskCache.synchronizedBuild$default(Unknown Source)
        at hydraulic.diskcache.LocalDiskCache.get(Unknown Source)
        at hydraulic.factory.tasks.cached.CachedBuildTask.load$hydraulic_factory(Unknown Source)
        ... 8 more

To submit a crash report, use --show-log=report.
Error: Process completed with exit code 1.
```
**To Reproduce**
This error occurs after the initial successful build and caching. It starts failing on the second build attempt.

**Comment by guiyanakuang** [comment]
Adding --show-log also fails to find the log
```
Sending crash report...
java.io.FileNotFoundException: /home/runner/.cache/hydraulic/conveyor/logs/log.latest.txt not found
	at java.net.http/jdk.internal.net.http.RequestPublishers$FilePublisher.create(RequestPublishers.java:289)
	at java.net.http/java.net.http.HttpRequest$BodyPublishers.ofFile(HttpRequest.java:704)
	at hydraulic.app.impl.AppFrameworkInstance.sendCrashReport(Unknown Source)
	at hydraulic.app.impl.AppFrameworkInstance.execute(Unknown Source)
	at picocli.CommandLine.execute(Unknown Source)
	at hydraulic.app.impl.AppFrameworkRuntime.init(Unknown Source)
	at hydraulic.conveyor.app.ConveyorApplication.main(Unknown Source)
To submit a crash report, use --show-log=report.
```

**Comment by mikehearn** [comment]
Thanks for the bug report. From looking, there might be a problem with our GitHub Action and how it manages the cache. Will test out a fix soon.

For the missing logs, it's a bit unclear what's going on. Probably that was from a second job? The GitHub Action should probably submit crash logs by default. We'll make that change.

**Comment by mikehearn** [comment]
This is actually a bug in GitHub Actions/Toolbox but it seems they aren't maintaining the relevant module properly (>100 open pull requests!). We'll work around it instead.

**Comment by mikehearn** [comment]
This should be fixed in 14.3
